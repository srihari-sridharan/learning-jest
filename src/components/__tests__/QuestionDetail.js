import { mapStateToProps, QuestionDetailDisplay } from "../QuestionDetail";
import renderer from 'react-test-renderer';
import React from 'react';

describe('The Question Detail Component', () => {
    describe('The container element', () => {
        describe('mapStateToProps', () => {
            it('should map to the state to props correctly', () => {
                const sampleQuestion = {
                    question_id: 42,
                    body: 'This is a sample questions'
                };
                const appState = {
                    questions: [sampleQuestion]
                };
                const ownProps = { question_id: 42 };
                const componentState = mapStateToProps(appState, ownProps);
                expect(componentState).toEqual(sampleQuestion);
            });
        });
    });

    describe('The display element', () => {
        it('should not regress', () => {
            const tree = renderer.create(
                <QuestionDetailDisplay
                    title="How do you learn Jest?"
                    body="1"
                    answer_count={0}
                    tags={['React', 'Jest', 'JavaScript']}
                />);

            expect(tree.toJSON()).toMatchSnapshot();
        });
    });
});