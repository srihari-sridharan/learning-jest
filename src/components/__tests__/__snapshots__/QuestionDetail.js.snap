// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Question Detail Component The display element should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    How do you learn Jest?
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          React
        </code>
        <code>
          Jest
        </code>
        <code>
          JavaScript
        </code>
      </div>
    </div>
    <div>
      <p>
        1
      </p>
    </div>
    <div>
      0
       Answers
    </div>
  </div>
</div>
`;
